node/malus/src/variants/common.rs

```diff
@@ -293,27 +364,80 @@ where
 								),
 							})
 						}
-						self.send_validation_response(
-							candidate_receipt.descriptor,
-							subsystem_sender.clone(),
-							response_sender,
-						);
-						None
+						// If the `PoV` is malicious, back the candidate with some probability `p`,
+						// which defaults to 100% for suggest-garbage-candidate variant.
+						let behave_maliciously = self.distribution.sample(&mut rand::thread_rng());
+						match behave_maliciously {
+							true => {
+								gum::info!(
+									target: MALUS,
+									"😈 Malicious behaviour is sampled as: {:?}. Backing candidate with malicious PoV.",
+									&behave_maliciously,
+								);
+
+								self.send_validation_response(
+									candidate_receipt.descriptor,
+									subsystem_sender.clone(),
+									response_sender,
+								);
+								None
+							},
+							// If the `PoV` is malicious, we behave normally with some probability `(1-p)`
+							false => Some(FromOrchestra::Communication {
+								msg: CandidateValidationMessage::ValidateFromChainState(
+									candidate_receipt,
+									pov,
+									timeout,
+									response_sender,
+								),
+							}),
+						}
 					},
 					FakeCandidateValidation::BackingInvalid |
 					FakeCandidateValidation::BackingAndApprovalInvalid => {
-						let validation_result =
-							ValidationResult::Invalid(self.fake_validation_error.clone().into());
-						gum::debug!(
-							target: MALUS,
-							para_id = ?candidate_receipt.descriptor.para_id,
-							"ValidateFromChainState result: {:?}",
-							&validation_result
-						);
-
-						// We're not even checking the candidate, this makes us appear faster than honest validators.
-						response_sender.send(Ok(validation_result)).unwrap();
-						None
+						// We back a garbage candidate with some probability `p`,
+						// which defaults to 100% for suggest-garbage-candidate variant.
+						let behave_maliciously = self.distribution.sample(&mut rand::thread_rng());
+						match behave_maliciously {
+							true => {
+								gum::info!(
+									target: MALUS,
+									"😈 Malicious behaviour is sampled as: {:?}. Setting ValidationResult::Invalid for valid candidate.",
```

Comment:
Why not log the  `validation_result` which contains the configured `fake_validation_error` ?